localstack/services/moto.py

```diff
@@ -0,0 +1,249 @@
+
+    :param context: the request context
+    :return: the HttpResponse from moto
+    """
+    status, headers, content = dispatch_to_moto(context)
+
+    return HttpResponse(response=content, status=status, headers=headers)
+
+
+def MotoFallbackDispatcher(provider: object) -> DispatchTable:
+    """
+    Wraps a provider with a moto fallthrough mechanism. It does by creating a new DispatchTable from the original
+    provider, and wrapping each method with a fallthrough method that calls ``reqest`` if the original provider
+    raises a ``NotImplementedError``.
+
+    :param provider: the ASF provider
+    :return: a modified DispatchTable
+    """
+    table = create_dispatch_table(provider)
+
+    for op, fn in table.items():
+        table[op] = _wrap_with_fallthrough(fn)
+
+    return table
+
+
+def _wrap_with_fallthrough(handler: ServiceRequestHandler) -> ServiceRequestHandler:
+    def _call(context, req) -> ServiceResponse:
+        try:
+            # handler will typically be an ASF provider method, and in case it hasn't been implemented, we try to
+            # fall through to moto
+            return handler(context, req)
+        except NotImplementedError:
+            return proxy_moto(context)
+
+    return _call
+
+
+def dispatch_to_moto(context: RequestContext) -> MotoResponse:
+    """
+    Internal method to dispatch the request to moto without changing moto's dispatcher output.
+    :param context: the request context
+    :return: the response from moto
+    """
+    service = context.service
+    request = context.request
+
+    # hack to avoid call to request.form (see moto's BaseResponse.dispatch)
+    setattr(request, "body", request.data)
+
+    # this is where we skip the HTTP roundtrip between the moto server and the boto client
+    dispatch = get_dispatcher(service.service_name, request.path)
+
+    return dispatch(request, request.url, request.headers)
+
+
+def get_dispatcher(service: str, path: str) -> MotoDispatcher:
+    url_map = get_moto_routing_table(service)
+
+    if len(url_map._rules) == 1:
+        # in most cases, there will only be one dispatch method in the list of urls, so no need to do matching
+        rule = next(url_map.iter_rules())
+        return rule.endpoint
+
+    matcher = url_map.bind(config.LOCALSTACK_HOSTNAME)
+    endpoint, _ = matcher.match(path_info=path)
+    return endpoint
+
+
+@lru_cache()
+def get_moto_routing_table(service: str) -> Map:
+    """Cached version of load_moto_routing_table."""
+    return load_moto_routing_table(service)
+
+
+def load_moto_routing_table(service: str) -> Map:
+    """
+    Creates from moto service url_paths a werkzeug URL rule map that can be used to locate moto methods to dispatch
+    requests to.
+
+    :param service: the service to get the map for.
+    :return: a new Map object
+    """
+    # code from moto.server.create_backend_app
+    backend_dict: BackendDict = get_moto_backend(service)
```

Comment:
I still think we should wrap this. Why? Because we could implement our backends in `moto-ext` and have this nice seamless integration:

1. Competing backends, but we think we have a better one implemented in moto-ext, this would return our backend first.
2. Non-existing backend in moto, would bring our backend implementation.